README.md

```diff
@@ -1,7 +1,128 @@
-#call
-
-HTTP Router.
+# call - Simple HTTP Router
 
 [![Build Status](https://secure.travis-ci.org/hapijs/call.png)](http://travis-ci.org/hapijs/call)
 
-Lead Maintainer - [Eran Hammer](https://github.com/hueniverse)
+### Lead Maintainer - [Eran Hammer](https://github.com/hueniverse)
+
+## Introduction
+```call``` is a simple node.js HTTP Router. It is used by popular [hapi.js](https://github.com/hapijs/hapi) web framework. It implements predictable and easy to use routing. Even if it is designed to work with Hapi.js, you can still use it as independent component.
+
+## Getting Started
+
+### Installation
+Install ```call``` as a dev dependency:
+
+```
+npm install --dev call
+// Or if using Yarn
+yarn add call
+```
+
+### Basic usage
+
+``` javascript
+// Import ES2015 modules
+// Use babel if using ES2015 modules (node.js doesn't support ES modules as of version 7.9.0)
+import * as Call from 'call';   // If using ES2015 modules
+
+// Or if using commonjs
+// const Call = require('call');
+
+// Create new router
+const router = new Call.Router();
+
+// Add route
+router.add({ method: 'get', path: '/' }, { label: 'root-path' });
+
+// Add another route
+router.add({ method: 'post', path: '/users' }, 'route specific data');
+
+// Add another route with dynamic path
+router.add({ method: 'put', path: '/users/{userId}' }, () => { /* ...handler... */ });
+
+// Match route
+router.route('post', '/users');
+/* If matching route is found, it returns an object containing
+    {
+        params: {},                     // All dynamic path parameters as key/value
+        paramsArray: [],                // All dynamic path parameter values in order
+        route: 'route specific data';   // routeData
+    }
+*/
+
+
+// Match route
+router.route('put', '/users/1234');
+/* returns
+    {
+        params: { userId: '1234' },
+        paramsArray: [ '1234' ],
+        route: [Function]
+    }
+*/
+```
+
+## API Reference
+
+### **new Call.Router([options])**
+Constructor to create a new router instance. To create router.
+```javascript
+const router = new Call.Router(); // Must be called with 'new' operator
+```
+It also accepts options object that currently supports following options:
+
+```isCaseSensitive: true | false```:<br>
+It specifies if paths should be treated as case sensitive. If set to true, then ```/users``` and ```/USERS``` are considered two different paths. Default value is ```true```.
+
+### **router.add(config, [routeData])**
+This method add a new route to router. Everytime, a route is added, router's internal table is analyzed to find possible conflicting route. ```config``` object has following fields:<br>
```

Comment:
`Everytime a route is added the router will make sure there are no conflicting routes and will throw on a duplicate.` *